modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -296,6 +328,31 @@ public void callCouchbaseRestAPI(String url, String payload) throws IOException
         httpConnection.getResponseCode();
     }
 
+    @Override
+    @SneakyThrows
+    protected void containerIsCreated(String containerId) {
+        File tempFile = copyConfigFromContainer(containerId);
+        appendPortsToConfig(tempFile);
+        copyFileToContainer(MountableFile.forHostPath(tempFile.toPath()), STATIC_CONFIG_PATH);
+    }
+
+    private File copyConfigFromContainer(String containerId) throws IOException {
```

Comment:
I would suggest using InputStreams directly to avoid files operations:
```
try (TarArchiveInputStream tarInputStream = new TarArchiveInputStream(this.dockerClient.copyArchiveFromContainerCmd(getContainerId(), containerPath).exec())) {
            tarInputStream.getNextTarEntry();
            // tarInputStream is now containers the file's content
        }
```